connection_test.go

```diff
@@ -1260,6 +1260,64 @@ func TestLastActivityTimePings(t *testing.T) {
 	})
 }
 
+func TestSendBufferSize(t *testing.T) {
+	opts := testutils.NewOpts().SetSendBufferSize(512).SetSendBufferSizeOverrides([]SendBufferSizeOverride{
+		{"abc", 1024},
+		{"abcd", 2048},
+		{"xyz", 3072},
+	})
+	tests := []struct {
+		processName          string
+		expectSendChCapacity int
+	}{
+		{
+			processName:          "abc",
+			expectSendChCapacity: 1024,
+		},
+		{
+			processName:          "abcd",
+			expectSendChCapacity: 1024,
+		},
+		{
+			processName:          "bcd",
+			expectSendChCapacity: 512,
+		},
+		{
+			processName:          "dabc",
+			expectSendChCapacity: 512,
+		},
+		{
+			processName:          "dabcd",
+			expectSendChCapacity: 512,
+		},
+		{
+			processName:          "abcde",
+			expectSendChCapacity: 1024,
+		},
+		{
+			processName:          "xyzabc",
+			expectSendChCapacity: 3072,
+		},
+	}
+	for _, tt := range tests {
+		testutils.WithTestServer(t, opts.NoRelay(), func(tb testing.TB, ts *testutils.TestServer) {
```

Comment:
you have `NoRelay` here. Can you move it up to the top where opts is declared, and add a comment about why we use `NoRelay`, e.g., `// Relay requires test logic to fork when checking incoming connections` or something